### PR TITLE
rbac: add some debug logging.

### PR DESCRIPTION
**Pull request by yangminzhu**
Signed-off-by: Yangmin Zhu <ymzhu@google.com>

*Description*: Added some log for help debugging
*Risk Level*: Low
*Testing*: n/a
*Docs Changes*: n/a
*Release Notes*: n/a

@lizan @mattklein123 @rodaine 
Thank you!

### CODE REVIEW

File: test/extensions/filters/http/rbac/rbac_filter_test.cc [hunk 1]
```diff
@@ -46,11 +46,11 @@ class RoleBasedAccessControlFilterTest : public testing::Test {
     filter_.setDecoderFilterCallbacks(callbacks_);
   }
 
-  void setDestinationPort(uint16_t port, int times = 2) {
+  void setDestinationPort(uint16_t port, int at_most_times = 3) {
     address_ = Envoy::Network::Utility::parseInternetAddress("1.2.3.4", port, false);
     auto& expect = EXPECT_CALL(connection_, localAddress());
-    if (times > 0) {
-      expect.Times(times);
+    if (at_most_times > 0) {
```
Comment (mattklein123) [inline test/extensions/filters/http/rbac/rbac_filter_test.cc]: This isn't a useful expectation. Please just make this an ON_CALL(...).WillByDefault(...)

File: source/extensions/filters/http/rbac/rbac_filter.cc [hunk 2]
```diff
@@ -65,13 +65,30 @@ RoleBasedAccessControlRouteSpecificFilterConfig::RoleBasedAccessControlRouteSpec
 
 Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::HeaderMap& headers,
                                                                       bool) {
+  ENVOY_LOG(
+      debug,
+      "checking request:\n"
```
Comment (mattklein123) [inline source/extensions/filters/http/rbac/rbac_filter.cc]: Debug lines in general should be single line. Please remove the newlines.

Comment (mattklein123) [inline source/extensions/filters/http/rbac/rbac_filter.cc]: "checking request" isn't a very descriptive message. Either make a new "rbac" log type, or prefix all of the messages you added with "rbac: " (or something similar).

File: test/extensions/filters/http/rbac/rbac_filter_test.cc (hunk 1 above)
Comment (yangminzhu) [reply test/extensions/filters/http/rbac/rbac_filter_test.cc]: Done

File: source/extensions/filters/http/rbac/rbac_filter.cc (hunk 2 above)
Comment (yangminzhu) [reply source/extensions/filters/http/rbac/rbac_filter.cc]: Done

Comment (yangminzhu) [reply source/extensions/filters/http/rbac/rbac_filter.cc]: Thanks for the suggestion, I added a new "rbac" log type.

File: source/common/common/logger.h [hunk 3]
```diff
@@ -45,7 +45,8 @@ namespace Logger {
   FUNCTION(upstream)             \
   FUNCTION(grpc)                 \
   FUNCTION(stats)                \
-  FUNCTION(thrift)
+  FUNCTION(thrift)               \
+  FUNCTION(rbac)
```
Comment (mattklein123) [inline source/common/common/logger.h]: please alpha order

Comment (yangminzhu) [reply source/common/common/logger.h]: Done.

**Pull request merged**